rubix-spi/src/main/java/com/qubole/rubix/spi/fop/ObjectPoolPartition.java

```diff
@@ -78,85 +82,55 @@ public void returnObject(Poolable<T> object)
 
   public Poolable<T> getObject(boolean blocking)
   {
-    if (objectQueue.size() == 0) {
-      // increase objects and return one, it will return null if pool reaches max size or if object creation fails
-      Poolable<T> object = increaseObjects(this.config.getDelta(), true);
-
-      if (object != null) {
-        return object;
-      }
-
-      if (totalCount == 0) {
-        // Could not create objects, this is mostly due to connection timeouts hence no point blocking as there is not other producer of sockets
-        throw new RuntimeException("Could not add connections to pool");
-      }
-      // else wait for a connection to get free
+    try {
+      takeSemaphore.tryAcquire(config.getMaxWaitMilliseconds(), TimeUnit.MILLISECONDS);
+    }
+    catch (InterruptedException e) {
+      e.printStackTrace();
+      throw new RuntimeException("Cannot get a free object from the pool");
     }
 
-    Poolable<T> freeObject;
-    try {
-      if (blocking) {
-        freeObject = objectQueue.take();
+    Poolable<T> object;
+    if (blocking) {
+      try {
+        object = objectQueue.take();
       }
-      else {
-        freeObject = objectQueue.poll(config.getMaxWaitMilliseconds(), TimeUnit.MILLISECONDS);
-        if (freeObject == null) {
-          throw new RuntimeException("Cannot get a free object from the pool");
-        }
+      catch (InterruptedException e) {
+        throw new RuntimeException(e); // will never happen
       }
     }
-    catch (InterruptedException e) {
-      throw new RuntimeException(e); // will never happen
+    else
+    {
+      object = tryGetObject(objectQueue.poll());
     }
-
-    freeObject.setLastAccessTs(System.currentTimeMillis());
-    return freeObject;
+    object.setLastAccessTs(System.currentTimeMillis());
+    return object;
   }
 
-  private Poolable<T> increaseObjects(int delta, boolean returnObject)
+  private Poolable<T> tryGetObject(Poolable<T> poolable)
```

Comment:
Dont take `Poolable` as arg, instead do `objectQueue.poll()` inside this method

Reply:
fixed.